panic: runtime error: kyanos/agent/protocol.MakeNewFakeData(...)

**Describe the bug**

Did not find a response deque extending the prior more to come response. requestID: 151663842
Did not find a response deque extending the prior more to come response. requestID: 151663856
panic: runtime error: index out of range [18] with length 18

goroutine 15 [running]:
kyanos/agent/protocol.MakeNewFakeData(...)
	/home/avalon/xiaoweihao/kyanos/agent/protocol/fakedata.go:17
kyanos/agent/conn.fillSyscallDataIfNeeded({0xc002318000, 0x7800, 0x7800}, 0xc010a56cb0, 0xc00f963880?)
	/home/avalon/xiaoweihao/kyanos/agent/conn/conntrack.go:527 +0x248
kyanos/agent/conn.(*Connection4).OnSyscallEvent(0xc003b8de00, {0xc002318000?, 0x1829fc1de89689f2?, 0x100000001?}, 0xc010a56cb0, 0xc000102540)
	/home/avalon/xiaoweihao/kyanos/agent/conn/conntrack.go:546 +0xf1
kyanos/agent/conn.(*Processor).processSyscallEvent(0xc1ea44701f36f4f0?, 0xc010a56cb0, 0xc000102540)
	/home/avalon/xiaoweihao/kyanos/agent/conn/processor.go:485 +0x2fd
kyanos/agent/conn.(*Processor).processOldSyscallEvents(0xc0001be480, 0xc000102540)
	/home/avalon/xiaoweihao/kyanos/agent/conn/processor.go:453 +0x112
kyanos/agent/conn.(*Processor).processTimedSyscallEvents(...)
	/home/avalon/xiaoweihao/kyanos/agent/conn/processor.go:441
kyanos/agent/conn.(*Processor).run(0xc0001be480)
	/home/avalon/xiaoweihao/kyanos/agent/conn/processor.go:315 +0xa46
created by kyanos/agent/conn.InitProcessorManager in goroutine 1
	/home/avalon/xiaoweihao/kyanos/agent/conn/processor.go:32 +0x171

**To Reproduce**
Steps to reproduce the behavior:

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**

![Image](https://github.com/user-attachments/assets/316eff35-f8b0-44a8-9597-e6bca05f46a5)

**Information (please complete the following information):**
 - OS: [e.g. use `uname -ar`]

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -162,6 +162,20 @@ jobs:
             sudo chmod +x /usr/local/bin/docker-compose
             docker-compose --version
 
+
+      - name: Test filter by l3/l4 info
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
+        with:
+          provision: 'false'
+          cmd: |
+            set -ex
+            uname -a
+            cat /etc/issue
+            if [ -f "/var/lib/kyanos/btf/current.btf" ]; then  
+                bash /host/testdata/test_filter_by_l4.sh 'sudo /host/kyanos/kyanos $kyanos_log_option --btf /var/lib/kyanos/btf/current.btf'
+            else  
+                bash /host/testdata/test_filter_by_l4.sh 'sudo /host/kyanos/kyanos $kyanos_log_option'
+            fi
       - name: Test Truncated Data parsing
         uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         if: ${{ !contains(fromJSON('["4.19-20240912.022020", "5.4-20240912.022020"]'), matrix.kernel) }}
@@ -329,21 +343,6 @@ jobs:
                 bash /host/testdata/test_base.sh 'sudo /host/kyanos/kyanos $kyanos_log_option --btf /var/lib/kyanos/btf/current.btf'
             else  
                 bash /host/testdata/test_base.sh 'sudo /host/kyanos/kyanos $kyanos_log_option'
-            fi
-
-
-      - name: Test filter by l3/l4 info
-        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
-        with:
-          provision: 'false'
-          cmd: |
-            set -ex
-            uname -a
-            cat /etc/issue
-            if [ -f "/var/lib/kyanos/btf/current.btf" ]; then  
-                bash /host/testdata/test_filter_by_l4.sh 'sudo /host/kyanos/kyanos $kyanos_log_option --btf /var/lib/kyanos/btf/current.btf'
-            else  
-                bash /host/testdata/test_filter_by_l4.sh 'sudo /host/kyanos/kyanos $kyanos_log_option'
             fi
 
       - name: Test kern evt

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -18,12 +18,12 @@ jobs:
   build:
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - uses: actions/checkout@v4.2.2
         with:
           submodules: recursive
 
       - name: Set up Go
-        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5
+        uses: actions/setup-go@v5.3.0
         with:
           go-version: '1.22.4'
 
@@ -48,7 +48,7 @@ jobs:
       #   run: make test
 
       - name: Store executable
-        uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4
+        uses: actions/upload-artifact@v4.6.1
         with:
           name: kyanos
           path: kyanos
@@ -80,12 +80,12 @@ jobs:
     timeout-minutes: 30
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - uses: actions/checkout@v4.2.2
         with:
           submodules: recursive
 
       - name: Retrieve stored kyanos executable
-        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4
+        uses: actions/download-artifact@v4.1.9
         with:
           name: kyanos
           path: kyanos

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -91,7 +91,7 @@ jobs:
           path: kyanos
 
       - name: Provision LVH VMs
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           test-name: kyanos-test
           image-version: ${{ matrix.kernel }}
@@ -116,7 +116,7 @@ jobs:
 
       - name: copy btf file
         if: ${{ startsWith(matrix.kernel, '4.') }}
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -131,7 +131,7 @@ jobs:
             # sudo cp $btf_file /var/lib/ptcpdump/btf/vmlinux
 
       - name: Install dependencies
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -163,7 +163,7 @@ jobs:
             docker-compose --version
 
       - name: Test Truncated Data parsing
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         if: ${{ !contains(fromJSON('["4.19-20240912.022020", "5.4-20240912.022020"]'), matrix.kernel) }}
         with:
           provision: 'false'
@@ -180,7 +180,7 @@ jobs:
             popd
 
       - name: Test Kafka
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -196,7 +196,7 @@ jobs:
             popd
 
       - name: Test CAP_BPF privilege check
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         if: ${{ !contains(fromJSON('["4.19-20240912.022020", "5.4-20240912.022020"]'), matrix.kernel) }}
         with:
           provision: 'false'
@@ -209,7 +209,7 @@ jobs:
             popd
 
       - name: Test CAP_SYS_ADMIN privilege check
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         if: contains(fromJSON('["4.19-20240912.022020", "5.4-20240912.022020"]'), matrix.kernel)
         with:
           provision: 'false'
@@ -222,7 +222,7 @@ jobs:
             popd
       
       - name: Test RocketMQ
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -238,7 +238,7 @@ jobs:
             popd
 
       - name: Test filter by comm
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -254,7 +254,7 @@ jobs:
             popd
 
       - name: Test gotls
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -270,7 +270,7 @@ jobs:
             popd
 
       - name: Test https
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -286,7 +286,7 @@ jobs:
             popd
 
       - name: Test side
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -302,7 +302,7 @@ jobs:
             popd
 
       - name: Test mysql
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -318,7 +318,7 @@ jobs:
             popd
 
       - name: Test base
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -333,7 +333,7 @@ jobs:
 
 
       - name: Test filter by l3/l4 info
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -347,7 +347,7 @@ jobs:
             fi
 
       - name: Test kern evt
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -361,7 +361,7 @@ jobs:
             fi
 
       - name: Test test docker filter by container id
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -375,7 +375,7 @@ jobs:
             fi
 
       - name: Test test docker filter by container name
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -389,7 +389,7 @@ jobs:
             fi
 
       - name: Test filter by pid
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -403,7 +403,7 @@ jobs:
             fi
 
       - name: Test test containerd filter by container name
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -417,7 +417,7 @@ jobs:
             fi
 
       - name: Test test containerd filter by container id
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -431,7 +431,7 @@ jobs:
             fi
 
       - name: Test redis
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |
@@ -446,7 +446,7 @@ jobs:
 
       - name: Test k8s
         if: ${{ startsWith(matrix.kernel, '6.') }}
-        uses: cilium/little-vm-helper@97c89f004bd0ab4caeacfe92ebc956e13e362e6b # v0.0.19
+        uses: cilium/little-vm-helper@c44c1221b104ee02ec0235211f7ace3c88eb11a2 # v0.0.19
         with:
           provision: 'false'
           cmd: |

--- a/agent/protocol/fakedata.go
+++ b/agent/protocol/fakedata.go
@@ -13,10 +13,10 @@ func MakeNewFakeData(size uint32) ([]byte, bool) {
 		copy(buf, fakeDataMarkPrefix)
 		size -= uint32(fakeDataMarkLen)
 		lenByte := []byte{byte(size >> 24), byte(size >> 16), byte(size >> 8), byte(size)}
-		buf[fakeDataMarkLen] = lenByte[0]
-		buf[fakeDataMarkLen+1] = lenByte[1]
-		buf[fakeDataMarkLen+2] = lenByte[2]
-		buf[fakeDataMarkLen+3] = lenByte[3]
+		buf[len(fakeDataMarkPrefix)] = lenByte[0]
+		buf[len(fakeDataMarkPrefix)+1] = lenByte[1]
+		buf[len(fakeDataMarkPrefix)+2] = lenByte[2]
+		buf[len(fakeDataMarkPrefix)+3] = lenByte[3]
 		return buf, true
 	}
 }
@@ -37,6 +37,6 @@ func getFakeDataSize(buf []byte, pos int) uint32 {
 	if len(buf) < fakeDataMarkLen {
 		return 0
 	} else {
-		return uint32(buf[fakeDataMarkLen])<<24 | uint32(buf[fakeDataMarkLen+1])<<16 | uint32(buf[fakeDataMarkLen+2])<<8 | uint32(buf[fakeDataMarkLen+3])
+		return uint32(buf[len(fakeDataMarkPrefix)])<<24 | uint32(buf[len(fakeDataMarkPrefix)+1])<<16 | uint32(buf[len(fakeDataMarkPrefix)+2])<<8 | uint32(buf[len(fakeDataMarkPrefix)+3])
 	}
 }

--- a/testdata/test_filter_by_l4.sh
+++ b/testdata/test_filter_by_l4.sh
@@ -24,7 +24,7 @@ function test_filter_by_remote_port() {
     remote_port=88
     timeout 20 ${CMD} watch --debug-output http --remote-ports "$remote_port" 2>&1  | tee "${LNAME_REMOTE_PORT}" &
     sleep 10
-    curl http://baidu.com || true
+    curl http://www.baidu.com || true
     wait
 
     cat "${LNAME_REMOTE_PORT}"
@@ -35,7 +35,7 @@ function test_filter_by_remote_port() {
     remote_port=80
     timeout 20 ${CMD} watch --debug-output http --remote-ports "$remote_port" 2>&1  | tee "${LNAME_REMOTE_PORT}" &
     sleep 10
-    curl http://baidu.com || true
+    curl http://www.baidu.com || true
     wait
 
     cat "${LNAME_REMOTE_PORT}"

--- a/testdata/test_filter_by_l4.sh
+++ b/testdata/test_filter_by_l4.sh
@@ -24,7 +24,7 @@ function test_filter_by_remote_port() {
     remote_port=88
     timeout 20 ${CMD} watch --debug-output http --remote-ports "$remote_port" 2>&1  | tee "${LNAME_REMOTE_PORT}" &
     sleep 10
-    curl http://baidu.com &>/dev/null || true
+    curl http://baidu.com || true
     wait
 
     cat "${LNAME_REMOTE_PORT}"
@@ -35,7 +35,7 @@ function test_filter_by_remote_port() {
     remote_port=80
     timeout 20 ${CMD} watch --debug-output http --remote-ports "$remote_port" 2>&1  | tee "${LNAME_REMOTE_PORT}" &
     sleep 10
-    curl http://baidu.com &>/dev/null || true
+    curl http://baidu.com || true
     wait
 
     cat "${LNAME_REMOTE_PORT}"

